Sileo/UI/PackageListViewController/PackageCell/PackageCollectionViewCell.swift

```diff
@@ -471,17 +503,49 @@ extension PackageCollectionViewCell: SwipeCollectionViewCellDelegate {
-        if let range: Range<String.Index> = conflicts.range(of: "firmware") {
-        let newconflicts = conflicts[(range.lowerBound ..< conflicts.endIndex)]
-        if let range2: Range<String.Index> = newconflicts.range(of: ">=") {
-            if let rangeofendfirmware: Range<String.Index> = newconflicts.range(of: ")") {
-            var firmwareinconflict = String(newconflicts[(range2.lowerBound ..< rangeofendfirmware.lowerBound)])
-            firmwareinconflict = firmwareinconflict.replacingOccurrences(of: ">=",with: "")
-            let number = (firmwareinconflict as NSString).floatValue
-            return number
+            if let range: Range<String.Index> = conflicts.range(of: "firmware") {
+            let newconflicts = conflicts[(range.lowerBound ..< conflicts.endIndex)]
+            if let range2: Range<String.Index> = newconflicts.range(of: ">") {
+                if let rangeofendfirmware: Range<String.Index> = newconflicts.range(of: ")") {
+                var firmwareinconflict = String(newconflicts[(range2.lowerBound ..< rangeofendfirmware.lowerBound)])
+                firmwareinconflict = firmwareinconflict.replacingOccurrences(of: ">",with: "").replacingOccurrences(of: "=",with: "")
+                let number = (firmwareinconflict as NSString).floatValue
+                return number
+                }
             }
         }
+    return 0.0
     }
-        return 0.0
+    
+    private func getFirmwareFromConflicts(conflicts: String) -> Float { //looking at this now, could have prob just used getFirmwareFromDepends and passed >/< into it but I already did this so eh
+            if let range: Range<String.Index> = conflicts.range(of: "firmware") {
+            let newconflicts = conflicts[(range.lowerBound ..< conflicts.endIndex)]
+            if let range2: Range<String.Index> = newconflicts.range(of: "<") {
+                if let rangeofendfirmware: Range<String.Index> = newconflicts.range(of: ")") {
+                var firmwareinconflict = String(newconflicts[(range2.lowerBound ..< rangeofendfirmware.lowerBound)])
+                firmwareinconflict = firmwareinconflict.replacingOccurrences(of: "<",with: "").replacingOccurrences(of: "=",with: "")
+                let number = (firmwareinconflict as NSString).floatValue
+                return number
+                }
+            }
+        }
+    return 0.0
+    }
+    
+    private func getConflictFirmwareIsEqual(conflicts: String) -> Bool {
+        guard let range = conflicts.range(of: "firmware") else {
+            return true
+        }
+        let newConflicts = conflicts[(range.lowerBound ..< conflicts.endIndex)]
+        
+        guard let secondRange = newConflicts.range(of: "("), let endRangeOfFirmware = newConflicts.range(of: ")") else {
+            return true
+        }
+        
+        if (String(newConflicts[(secondRange.lowerBound ..< endRangeOfFirmware.lowerBound)]).contains("=")) {
```

Comment:
Scrap this and do `return String(newConflicts[(secondRange.lowerBound ..< endRangeOfFirmware.lowerBound)]).contains("=")` instead

Reply:
just tested, and it works. any more requests or is rest good?

Reply:
Tons more requests but I’m not home atm, for one, the code looks super performance intensive